contracts/test/PoolFDTLosses.t.sol

```diff
@@ -0,0 +1,319 @@
+        // Verify the interest.
+        withinDiff(
+            getClaimedInterestNetOfFees(claimInfo[1], pool.delegateFee(), pool.stakingFee()),
+            usdc.balanceOf(address(lex)),
+            1
+        );
+
+        // Time warp to default
+        hevm.warp(custom_loan.nextPaymentDue() + globals.defaultGracePeriod() + 1);
+
+        // Setting slippage high enough to let it through the liquidations.
+        gov.setMaxSwapSlippage(7000);
+
+        // Pool Delegate trigger a default
+        pat.triggerDefault(address(pool), address(custom_loan), address(dlFactory));
+        
+        // Check for successful default.
+        assertTrue(uint8(custom_loan.loanState()) == 4, "Unexpected Loan state");
+
+        // PD claims funds and also sells the stake to recover the losses.
+        claimInfo = pat.claim(address(pool), address(custom_loan), address(dlFactory));
+        
+        assertTrue(claimInfo[6] > 0, "Loan doesn't has default suffered");
+
+        uint256 poolLosses = pool.poolLosses();
+        assertTrue(poolLosses > 0, "Pool losses should be greater than 0");
+        withinPrecision(pool.recognizableLossesOf(address(lex)), poolLosses, 5);
+
+        // Time warp to withdraw
+        hevm.warp(block.timestamp + pool.lockupPeriod() + 1);
+
+        // Fails to transfer if the losses are > 0.
+        assertTrue(
+            !lex.try_transfer(address(pool), address(leo), 1),
+            "Should not allowed to transfer because losses are > 0"
+        );
+
+        // Lex is withdrawing funds from the Pool with losses.
+        assertTrue(lex.try_intendToWithdraw(address(pool)), "Failed to intend to withdraw");
+
+        // Time warp to withdraw window
+        hevm.warp(block.timestamp + globals.lpCooldownPeriod() + 1);
+
+        // Before withdrawing funds.
+        uint256 old_lex_bal = usdc.balanceOf(address(lex));
+
+        assertTrue(lex.try_withdraw(address(pool), depositAmt), "Failed to withdraw");
+
+        assertEq(
+            (usdc.balanceOf(address(lex)) - old_lex_bal) / USD,
+            (depositAmt - poolLosses) / USD,
+            "New balance should count the losses"
+        );
+    }
+
+    function test_multipleLpBearingLossesWithMultiplePools(
+        uint256 apr,
+        uint256 index,
+        uint256 numPayments,
+        uint256 requestAmount
+    ) public {
+        uint256 minimumRequestAmt = PoolLib.getSwapOutValueLocker(address(bPool), USDC, address(pool.stakeLocker()));
+        minimumRequestAmt        += PoolLib.getSwapOutValueLocker(address(bPool), USDC, address(pool2.stakeLocker()));
+
+        // Create Loan and a deposit amount from arguments
+        (Loan custom_loan, uint256 totalDepositAmt) = getLoanAndDepositAmount(apr, index, numPayments, requestAmount, 0, minimumRequestAmt);
+
+        // Get 4 values each between 4 and 100 (thus, even a 4, 100, 100, 100 will not result in a zero after a division)
+        uint256 lex_depositAmt = constrictToRange(index, 4, 100, true);
```

Comment:
isn't it take a much wider range and all LPs deposits will be equal and there is no value to hash the index as it always gives you the same value because the index is not changing after every deposit.

Reply:
But the next line is `index = uint256(keccak256(abi.encodePacked(index)))`, so Lee, Leo, and Liz's deposit amounts will all be pseudo random.